src/AutoMerge/github.jl

```diff
@@ -87,6 +87,35 @@ function _get_updated_pull_request(pull_request::GitHub.PullRequest; auth::GitHu
     return updated_pr
 end
 
+function _issue_labels(pull_request::GitHub.PullRequest; auth::GitHub.Authorization)
+    pr_base_repo = base_repo(pull_request)
+    pr_number = number(pull_request)
+    issue = GitHub.issue(pr_base_repo, pr_number; auth=auth)
+    if issue.labels === nothing
+        labels = String[]
+    else
+        labels = [ d["name"] for d in issue.labels ]
+    end
+    return issue, labels
+end
+
+function add_label!(pull_request::GitHub.PullRequest, label::String; auth::GitHub.Authorization)
```

Comment:
Regarding the `!`: I tried to follow the example of `merge!`; it doesn't actually modify the `pull_request::GitHub.PullRequest ` object itself, but it does modify the pull request on GitHub.

Reply:
Yeah in this codebase I used `!` as a reminder to myself that it modifies some GitHub object.